transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -390,8 +390,8 @@ protected void run() {
                     //increase the size of the array as we needed the whole space for the events
                     events.increase();
                 }
-            } catch (Throwable t) {
-                handleLoopException(t);
+            } catch (Exception e) {
+                handleLoopException(e);
```

Comment:
why this change ? Thins looks not right...

Reply:
Sorry, I should have mentioned this in the original comment. We also have a need to be able to kill threads using Thread.stop(). Under the hood, Thread.stop actually kills threads using ThreadDeathError. Unfortunately, this means that if you swallow all Throwables, the thread becomes unkillable.

Reply:
Why you just cant normally shutdown the `EventLoop` ? Like `EventLoopGroup.shutdownGracefully().sync()` ?

Reply:
This is for infrastructure code that doesn't necessarily have knowledge about when EventLoops are created. Our clients are free to create them as they please and we need a generic, scalable solution for taking them down as quickly as possible.

Reply:
Hmm ... Thread.stop() was removed some time ago . How will think work for you with newer JDK versions ?

Reply:
Was it removed? In the version we use, it is still just deprecated. We might be petitioning the OpenJVM maintainers to keep it around or forking the JVM internally just for that purpose.

Reply:
Yes in JDK11: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8204243

Reply:
Oh, this is for the official JDK though, right? We are using OpenJDK11.

In either case, I see that our motivation for this change is rather unique to our circumstances. Is there a specific reason to swallow all Throwables here? Errors are supposed to be so severe that no application could reasonably handle them (like OutOfMemory, JVM errors, etc).

Reply:
As far as I know it’s the same for OpenJDK... I guess we could rethrow the ThreadDeath tho

Reply:
That works for me. How does this most recent commit look?